controllers/API/announcement.js

```diff
@@ -15,38 +14,43 @@ router.put("/API/announcement", auth.isAuthorised("ALTER_ANNOUNCEMENTS"), functi
 		}
 	}, function (err) {
 		if (err) {
-			logger.warning("Unable to edit announcement.\n" + err);
-			res.sendStatus(400);
+			err.shouldReload = true;
```

Comment:
If I am not mistaken, we could create our own error, such as `ApiError` and throw these in our endpoints.

Then, it should be automatically passed to an error handler. So instead of doing 
```javascript
err.shouldReload = true;
err.status = 400;
next(err);
```
We could do something like:
`throw new ApiError()`
and then we handle it in an error handler.

Reply:
I will add that to my todo list seems more reasonable to have a custom class.